protos/escrow/escrow.proto

```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
+import "google/protobuf/timestamp.proto";
+
+package btfs;
+
+service EscrowService{
+	rpc SubmitContracts(EscrowContractRequest) returns (EscrowContractResult) {}
+	rpc PayIn(PayinRquest) returns (PayinResult) {}
+	rpc CheckPayment(ContractID) returns (PayinStatus) {}
+	rpc CancelContracts(CancelContractRequest) returns (CancelContractResult) {}
+};
+
+enum TokenType {
+	BTT = 0;
+}
+
+enum Schedule {
+	MONTHLY = 0;
+	QUARTERLY = 1;
+	ANNUALLY = 2;
+}
+
+message ContractID {
+	bytes contract_id = 1;
+}
+
+message EscrowContract {
+	bytes contract_id = 1; // identity of a contract
+	bytes buyer_address = 2;
+	bytes buyer_signature = 3;
+	bytes seller_address = 4;
+	bytes seller_signature = 5; // the seller signature
+	bytes auth_address = 6; // authority who can cancel escrow
+	bytes auth_signature = 7;
+	int64 amount = 8;
+	int64 collateral_amount = 9;
+	int64 withhold_amount = 10; // escrow hold amount till the last layment
+	TokenType token_type = 11; // token type enum
+	Schedule payout_schedule = 12; // schedule enum
+	int32 num_payouts = 13; // numbers of payouts
+	bytes reference = 14; // reference like hash value etc
+}
+
+message EscrowContractRequest {
+	repeated EscrowContract contract = 1;
+}
+
+message Escrow {
+	bytes contract_id = 1;
+	int64 escrow_id = 2;
+	bytes from_address = 3;
+	bytes to_address = 4;
+	int64 amount = 5;
+	int64 collateral_amount = 6;
+}
+
+message EscrowContractResult {
+	bytes payin_id = 1; // for buyer to pay in a group of escrows
+	repeated Escrow escrow = 2;
+	bytes escrow_address = 3;
+	google.protobuf.Timestamp escrow_signed_time = 4;
+	bytes escrow_signature = 5;
+}
+
+message PayinRquest {
+	bytes payin_id = 1;
```

Comment:
Suggest add amount here, to prove renter is full of sure how much he will pay.

Reply:
Request time was suggest to add too, with this can help us to provide some auditing information.

Reply:
amount is not necessary, escrow only follow the amount in contract. Otherwise if renter put a smaller amount, what's the expectation for escrow to do.
Payin request time is escrow received time to make interface API simple.

Reply:
Yes, the amount is not necessary here from process view, escrow service already know the amount in the previous submit contract. But from verification view, as renter cannot be trust throughly, suggest to add the field here so that we can prove that while renter submit the payin request, he fully sure for the amount he will pay.

Reply:
once contract signed, renter is sure for the amount, no need to add the amount here.

Reply:
talked to Junlin, nice to have, so not add now.